add-on/src/recovery/recovery.js

```diff
@@ -0,0 +1,66 @@
+'use strict'
+/* eslint-env browser, webextensions */
+
+import choo from 'choo'
+import html from 'choo/html/index.js'
+import { i18n, runtime } from 'webextension-polyfill'
+import { renderLogo } from '../landing-pages/welcome/page.js'
+import createWelcomePageStore from '../landing-pages/welcome/store.js'
+import './recovery.css'
+
+const app = choo()
+
+// TODO (whizzzkid): refactor base store to be more generic.
+app.use(createWelcomePageStore(i18n, runtime))
+// Register our single route
+app.route('*', (state) => {
+  const { hash } = window.location
+  const { href: publicURI } = new URL(decodeURIComponent(hash.slice(1)))
+  if (!publicURI) {
+    return
+  }
+
+  const openURLFromHash = () => {
+    try {
+      console.log('Opening URL from hash:', publicURI)
+      window.location.href = publicURI
+    } catch (err) {
+      console.error('Failed to open URL from hash:', err)
+    }
+  }
+
+  // if the IPFS node is online, open the URL from the hash, this will redirect to the local node.
+  if (state.isIpfsOnline) {
+    openURLFromHash()
+    return
+  }
+
+  return html`<div class="recovery-root flex justify-center items-center">
+    <div class="flex w-70">
+      <div class="ma2 flex w-third flex-column transition-all">
+        ${renderLogo(false, 256)}
+      </div>
+      <div class="ma2 flex flex-column transition-all">
+        <h1 class="f3 fw6">${i18n.getMessage('recovery_page_sub_header')}</h1>
+        <p class="f5 fw4">${i18n.getMessage('recovery_page_message')}</p>
+        <p class="f5 fw4"><span class="b-ns">Public URI:</span> <a href="${publicURI}">${publicURI}</a></p>
+        <button
+          class="fade-in w-50 ba bw1 b--aqua bg-aqua snow f7 ph2 pv3 br4 ma1 pointer"
```

Comment:
`aqua`  has bad contrast, we fixed this in webui by switching to teal – see https://github.com/ipfs/ipfs-webui/issues/1379

> ![2023-01-11-173344_315x87_scrot](https://user-images.githubusercontent.com/157609/211862703-ec993062-625b-48de-a01d-907920ab0850.png)
> ![2023-01-11-173451_303x77_scrot](https://user-images.githubusercontent.com/157609/211862923-f3967770-191d-47e9-b87b-cb4356023fde.png)



```suggestion
          class="fade-in w-50 ba bw1 b--teal bg-teal snow f7 ph2 pv3 br4 ma1 pointer"
```